### PR TITLE
Basic Skeleton for the Complaints form

### PR DESCRIPTION
**Pull request by bendiggle**
Basic Skeleton for the Complaints for using the hof-bootstrap.
Added apps directory which includes fields, translations, steps, etc.
Added Assets.
Added test directory.
Updated app.js from hof-bootstap

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+const bootstrap = require('hof-bootstrap');
+const steps = require('./apps/steps.js');
+
+bootstrap({
+  routes: [{
+    steps: steps
```
Comment (bendiggle) [inline app.js]: @JoeChapman Is this the correct way to use the bootstrap?

File: apps/fields/index.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+'use strict';
```
Comment (bendiggle) [inline apps/fields/index.js]: @JoeChapman @joefitter @gxxm @easternbloc How do you guys feel about the structure being '/apps/fields/.....' or do you think it should be 'apps/complaints/fields/....'?

File: LICENSE [hunk 3]
```diff
@@ -0,0 +1,674 @@
+                    GNU GENERAL PUBLIC LICENSE
```
Comment (easternbloc) [inline LICENSE]: we use v2 not v3

Comment (easternbloc) [inline LICENSE]: and tbh it's about time we questioned if this can be MIT

Comment (bendiggle) [inline LICENSE]: @easternbloc I'll change it to v2 now but I copied this from the hof-boilerplate so will need changing there too. 
I think that has  been mentioned before too but to be honest I have no clue how licensing works or how you do it!

**Comment by joefitter** [comment]
A lot of changes have been made to GRO/ROTM/Bootstrap since this skeleton was put together, I would be tempted to try and build UKVI complaints using bootstrap to save repeating ourselves again

File: apps/fields/index.js (hunk 2 above)
Comment (joefitter) [inline apps/fields/index.js]: neither, it should be /src/fields IMO

Comment (joefitter) [inline apps/fields/index.js]: apps suggests there is gunna be multiple apps

**Comment by joefitter** [comment]
ignore me - just seen bootstrap 💃

**Comment by bendiggle** [comment]
@JoeChapman @joefitter @gxxm @easternbloc I have renamed 'apps/' to 'src/' and have updated the License to v2

File: app.js (hunk 1 above)
Comment (joefitter) [inline app.js]:
```js
{
  steps,
  ...
}
```
 - shorthand for `steps: steps`. No biggie, just a fyi

File: test/.eslintrc [hunk 4]
```diff
@@ -0,0 +1,6 @@
+extends:
+  - '../.eslintrc'
+  - 'homeoffice/config/testing'
+
+env:
```
Comment (joefitter) [inline test/.eslintrc]: env isn't needed if we are extending ../.eslintrc if it is defined there - which I assume it is as silly to have tests in es6 but src in es5.

Actually, why is this file even here? Does dotfiles not include test/.eslintrc @JoeChapman ?

Comment (bendiggle) [inline test/.eslintrc]: I'm not sure dotfiles will do this but @JoeChapman would know better. I'll remove the env now

File: app.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+const bootstrap = require('hof-bootstrap');
+const steps = require('./apps/steps.js');
```
Comment (JoeChapman) [inline app.js]: but if you rename `apps` to `src` then `steps.js` won't be found

File: package.json [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    "hof-bootstrap": "^0.1.1",
+    "hof-dotfiles": "^2.0.0",
+    "hogan-express-strict": "^0.5.4",
+    "hogan.js": "^3.0.2",
+    "i18n-lookup": "^0.1.0",
+    "jquery": "^2.1.4",
+    "lodash": "3.0.1",
+    "node-sass": "^3.7.0",
+    "redis": "^0.12.1",
+    "winston": "^1.0.1",
```
Comment (JoeChapman) [inline package.json]: Most of these deps are included in hof-bootstrap

File: app.js (hunk 5 above)
Comment (joefitter) [inline app.js]: automagical lookup eh? sounds.... magical

File: src/fields/index.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+'use strict';
+
+module.exports = {
+
```
Comment (JoeChapman) [inline src/fields/index.js]: Don't think `src` as the location is going to work?

File: src/steps.js [hunk 8]
```diff
@@ -0,0 +1,5 @@
+'use strict';
+
+module.exports = {
+
```
Comment (JoeChapman) [inline src/steps.js]: Don't think `src` as the location is going to work?

Comment (joefitter) [inline src/steps.js]: my bad, I thought bootstrap was more configurable than that. I'll add an issue

Comment (JoeChapman) [inline src/steps.js]: yeah, at the moment, it expects it to be `apps/`

File: app.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+const bootstrap = require('hof-bootstrap');
+const steps = require('./apps/complaints/steps.js');
+
+bootstrap({
+	views: false,
+	fields: false,
+	routes: [
+		steps
```
Comment (JoeChapman) [inline app.js]: This isn't really a step. Routes is an array of route objects. Each route object has a `steps` field, but steps are not routes.

File: apps/complaints/steps.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+			template: 'step',
+			fields: ['name-text'],
+			next: '/step1'
+		},
+		'/step1': {
+			template: 'step',
+			fields: ['name-text']
+		}
+	}
+};
```
Comment (JoeChapman) [inline apps/complaints/steps.js]: See note above about steps and routes. This is actually a route with a `steps` field.

**Comment by JoeChapman** [comment]
LGTM

**Comment by bendiggle** [comment]
@JoeChapman yeah, makes sense. I've updated it now

**Comment by JoeChapman** [comment]
@joefitter @easternbloc @gxxm?

**Comment by gxxm** [comment]
:thumbsup: :thumbsup:

**Pull request merged**